### DISCUSSION TITLE
Feature suggestions

### DISCUSSION THREAD

**Discussion by AlwaysNewDepths** [xFutte/futte-newspaper#21] [Ideas]
Thanks for this great script. Having a fully functional in-city newspaper really helps reporters with their RP and the city as a whole to find out information in a timely manner. Such a high quality feel to it too.

I have a couple of feature suggestions you might like to consider.

* Ability to re-order articles.
* Ability to "pin" articles to the top of the list.
* Make newspaper items decay so news is "out of date" and item is unusable after a specified timeframe, forcing citizens to buy a new newspaper to get up to date information.

Thanks again for your brilliant work.

**Comment by xFutte** [comment]
Thanks for the great feedback. I'm happy that you and your community are enjoying the script!

Great suggestions. Re-ordering and pinning would be great, so I'll look into that when I find the time. In regards to the decay, I'd rather implement a limit that the newspaper display news to no later than the date the newspaper was bought. Then decay of the inventory item itself can be added, from another script, which also includes decay of food, drinks, etc.

**Reply by AlwaysNewDepths** [reply]
Sounds great. Thanks for the reply and I look forward to seeing how the script evolves in the future

**Comment by AlwaysNewDepths** [comment]
Another one I've thought of - The ability to make an article inactive, so keeping it in the list for later usage, but making it inactive so people cannot see it in the list.
This would be useful for regular articles of information, or writing drafts of articles and finishing/publishing them later on.

**Comment by AlwaysNewDepths** [comment]
I had an email to say you'd mentioned another feature about a server update here but I can't see it to reply to it for some reason.

I can see how it would be useful in some GTA RP communities, especially public servers. In my particular community we try to keep all of that information in the Discord as technically it's OOC and not RP information. So, in my opinion, it's a good feature to add but personally I'd like to be able to toggle the page on or off in the config or reporter actions tab.

**Reply by xFutte** [reply]
Yeah, deleted it by accident.

As you state, lots of people might keep wanting it in Discord. That is why I initially didn't implement it completely but still added the option for it to be added in the future.

**Comment by AlwaysNewDepths** [comment]
Another built in feature could be to have the option to specify a bank account for the newspaper money to go into. This could either be a particular reporter role's account or a separate "news company" account which likely already exists (as is setup in the city I play in).

**Reply by xFutte** [reply]
I'm not 100% sure how that would be done (for now), so I'll start with some of the suggestions you've mentioned above. I really like some of them.

**Comment by AlwaysNewDepths** [comment]
Another suggestion for you.

Being able to post a YouTube link (video or live) allows it to be watched within the article.
This would allow reporters to "go live" with a TV broadcast from within the newspaper.
A bit meta and unrealistic maybe, but a good bit of functionality to have. It means all news content could be consumed within RP rather than having to go to a YouTube window.
Either within the article, or maybe as a separate tab along the top which could link to "video articles" that can be posted individually.

**Reply by xFutte** [reply]
Thanks. Might look into it, but some of the other suggestions I'd like to implement first, when I get to it.

**Reply by AlwaysNewDepths** [reply]
Yes, of course. I'm just adding ideas as I think of them. No pressure to actually do anything with them, just floating ideas around.
